### PR TITLE
Update CDT Cloud to 1.38 

### PR DESCRIPTION
**Pull request by ndoschek**
#### What it does

- Updates CDT.cloud Blueprint to the latest master of Theia Blueprint ([5fe2d3](https://github.com/eclipse-theia/theia-blueprint/commit/5fe2d301210a4e4fdde8e5eb91a25988214cacbf)).
- Update Theia versions to 1.38.0 
  - Align package versions to 1.38.0 as well
  - Add missing version tags in browser and docker package.json (to reduce build warnings)
  - Align default configuration (i.e. applicationName and preferences) for applications
  - Update resolutions block
- Update theiaPlugins to latest versions 
  - Update theia-traceviewer to 0.2.0-next.20230619144800.297af68.0
  - Update eclipse-theia.builtin-extension-pack to 1.77.0
  - Update cdt-gdb-vscode to 0.0.99
  - Update vscode.builtin-cpp to 1.77.0
  - Update vscode-clangd to 0.1.24
  - Update vscode-cmake to 1.14.33
  - Align ignored theia plugins extensions
  - Remove explicit adding of vscode.markdown-language-features, as it is included in builtin-extension-pack
- Re-add `request` dependency for tracecompass-server-download

Contributed on behalf of STMicroelectronics

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

### CODE REVIEW

File: theia-extensions/theia-blueprint-product/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   },
   "theiaExtensions": [
     {
-      "frontendElectron": "lib/browser/theia-blueprint-frontend-module",
+      "frontend": "lib/browser/theia-blueprint-frontend-module",
       "electronMain": "lib/electron-main/theia-blueprint-main-module",
       "frontend": "lib/browser/theia-blueprint-frontend-module",
```
Comment (sgraband) [inline theia-extensions/theia-blueprint-product/package.json]: This now contains the same frontend entry twice. Is this correct?

**Review by sgraband — CHANGES_REQUESTED** [review]
Thank you @ndoschek! Changes look good and work mostly great. I just have one inline comment and an issue with the trace viewer. It seems like i cannot launch the trace server from the application itself. Does this work for you?

Adjusting the "trace Viewer.trace Server.path" setting and the trigger the `Start Trace Server` command (or by clicking `Open Trace` in the Trace Viewer View), gives me an error that the trace server cannot be found at the path. (I am pretty certain, that the path is correct ;)) Could you check that?

**Comment by sgraband** [comment]
The last commit is missing the `Contribution on behalf` message.

File: theia-extensions/theia-blueprint-product/package.json (hunk 1 above)
Comment (ndoschek) [reply theia-extensions/theia-blueprint-product/package.json]: I just checked, there was a mistake in merging the remote branch, after that the frontend entry was listed twice, but I resolved that with commit [`Update Theia versions to 1.38.0`](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/pull/59/commits/42bfa9583be7119246a55af585e4a0e1ede8a235).

**Comment by ndoschek** [comment]
Hi again @sgraband,
I double checked and the trace server works for me in both the browser and electron use case.
I downloaded the [zipped trace server](https://download.eclipse.org/tracecompass.incubator/trace-server/rcp/trace-compass-server-latest-linux.gtk.x86_64.tar.gz) and then set the path to the tracecompass-server executable (e.g. `/home/nina/Clients/OpenSource/CDTCloud/tmp-tracecompass/trace-compass-server-latest-linux.gtk.x86_64/trace-compass-server/`).

**Comment by bhufmann** [comment]
Thanks for the upgrade. Shouldn't the upgrade go to the latest [Theia community release](https://theia-ide.org/releases/) first which 1.37.x instead of the latest Theia release 1.38.x? 

Please let me know.

**Comment by sdirix** [comment]
@bhufmann is right, in general we want to base on the latest community release, which was 1.37.

However Theia Blueprint itself is on a monthly schedule anyway and already available for 1.38 (the links to older versions are not easy to find to download). As the update work was already done I would be fine with skipping the community release this time and going straight to 1.38. Then we return to community release based releases with the next community release.

@bhufmann Is there some specific constraint for which you would like to see CDT Cloud Blueprint to stay on 1.37 instead of 1.38? If there is a specific technical constraint so that for example trace compass works better with 1.37 then we should go to that version instead.

**Comment by sdirix** [comment]
@bhufmann Thanks for the uplift. I think it's worth it already simply to list it at https://theia-ide.org/releases/ :+1: 

If the 1.37 version of trace compass works with Theia 1.38 without problems then I would now keep it that way. For next time we'll be more careful to correctly align with the community release again.

**Comment by ndoschek** [comment]
@sdirix
I ran the tests locally, and they work for me (on linux) without any issues. Could you please trigger the build checks again? TIA

**Review by sgraband — APPROVED** [review]
Thank you @ndoschek! LGTM :+1:

**Pull request merged**